python/coinbase-agentkit/coinbase_agentkit/wallet_providers/smart_wallet_provider.py

```diff
@@ -0,0 +1,165 @@
+
+        if config.cdp_api_key_name and config.cdp_api_key_private_key:
+            Cdp.configure(
+                api_key_name=config.cdp_api_key_name,
+                private_key=config.cdp_api_key_private_key.replace("\\n", "\n")
+            )
+        else:
+            Cdp.configure_from_json()
+
+        if config.smart_wallet_address:
+            self._smart_wallet = to_smart_wallet(
+                signer=config.signer, smart_wallet_address=config.smart_wallet_address
+            )
+        else:
+            self._smart_wallet = SmartWallet.create(config.signer)
+
+        self._smart_wallet = self._smart_wallet.use_network(
+            chain_id=int(self._network.chain_id),
+            paymaster_url=config.paymaster_url,
+        )
+
+    def get_address(self) -> str:
+        """Get the smart wallet address."""
+        return self._smart_wallet.address
+
+    def get_network(self) -> Network:
+        """Get the network information."""
+        return self._network
+
+    def get_name(self) -> str:
+        """Get the name of the wallet provider."""
+        return "cdp_smart_wallet_provider"
+
+    def sign_message(self, message: str | bytes) -> HexStr:
+        """Smart wallets cannot sign raw messages.
+
+        Raises:
+            NotImplementedError: Always, since smart wallets do not support signing raw messages.
+
+        """
+        raise NotImplementedError("Smart wallets do not support signing raw messages.")
+
+    def sign_typed_data(self, typed_data: dict[str, Any]) -> HexStr:
+        """Smart wallets cannot sign typed data.
+
+        Raises:
+            NotImplementedError: Always, since smart wallets do not support signing typed data.
+
+        """
+        raise NotImplementedError("Smart wallets do not support signing typed data.")
+
+    def sign_transaction(self, transaction: TxParams) -> SignedTransaction:
+        """Smart wallets cannot sign transactions.
+
+        Raises:
+            NotImplementedError: Always, since smart wallets do not support signing transactions.
+
+        """
+        raise NotImplementedError("Smart wallets do not support signing transactions.")
+
+    def send_transaction(self, transaction: TxParams) -> HexStr:
+        """Send a transaction using the smart wallet.
+
+        Unlike traditional Ethereum transactions, this method submits a User Operation
+        instead of directly broadcasting a transaction. The smart wallet handles execution,
+        but a standard transaction hash is still returned upon completion.
+        """
+        user_operation = self._smart_wallet.send_user_operation(
+            calls=[
+                EncodedCall(to=transaction["to"], data=transaction.get("data", b""), value=transaction.get("value", 0)),
+            ]
+        )
+        result = user_operation.wait()
+        if result.status == UserOperation.Status.COMPLETE:
```

Comment:
As of `@coinbase/coinbase-sdk` 0.21.0 (which goes out tomorrow), this line will change to `if result.status == "complete"`

I propose a fast-follow